docs: translate docs/deployment.md (#62)
Nextjs-ja-translation/Nextjs-ja-translation-docs: docs/deployment.md

---

> description: Next.jsアプリをVercelやその他のホスティングオプションを使って本番環境にデプロイします。

Reviewer: ```suggestion
description: Next.js アプリを Vercel やその他のホスティング方法を使って本番環境にデプロイします。
```
「ホスティングオプション」についてですが、ここで指しているのは「Vercel以外のホスティング方法（サービス）の選択肢」だと思いますので、単にオプションとしてしまうと「他の方法がある」というニュアンスが弱いと思いました。いかがでしょうか？

Reviewer 2: ご指摘ありがとうございます！
おっしゃる通りだと思います。修正を取り込ませていただきます。

---

# デプロイ

> ## Vercel (推奨する方法)

Reviewer: ```suggestion
## Vercel (推奨)
```
単に推奨で良いかと思います！

Next.jsを本番環境にデプロイする最も簡単な方法は、Next.jsの開発者が提供する**[Vercel platform](https://vercel.com)**を利用することです。[Vercel](https://vercel.com)は、静的サイト・JAMstackなアーキテクチャでのデプロイ、サーバーレスファンクション機能をサポートするグローバルCDNを備えたオールインワンプラットフォームです。

### はじめに

> はじめに、Next.js アプリを任意の Git プロバイダーにプッシュしてください。[GitHub](http://github.com/)、[GitLab](https://gitlab.com/)、[BitBucket](https://bitbucket.org/)などです。リポジトリはプライベートまたはパブリックでも構いません。

Reviewer: ```suggestion
はじめに、Next.js アプリを任意の Git プロバイダーにプッシュしてください。[GitHub](http://github.com/)、[GitLab](https://gitlab.com/)、[BitBucket](https://bitbucket.org/)などです。リポジトリはプライベートとパブリックのどちらでも構いません。
```
「プライベートまたはパブリックでも構いません」という言い回しが気になり修正してみましたがいかがでしょうか？

> 次に、以下の手順に従ってください。

Reviewer: ```suggestion
次に、以下の手順に従ってください:
```
記載がなく申し訳ございません🙏　原文にコロンがある場合は訳文もコロンとさせて下さい！

> 1. [Vercelにサインアップする](https://vercel.com/signup) （クレジットカードの登録は不要）

Reviewer: ```suggestion
1. [Vercelにサインアップする](https://vercel.com/signup) （クレジットカードの登録は不要）。
```
細かいですが、原文にピリオドがあるので訳文にも句点にも記載することとさせてください！

2. サインアップ後に、[「Import Project」](https://vercel.com/import)ページに遷移します。「From Git Repository」でデプロイに使用する Git プロバイダーを選択し、設定をします。(設定は各プロバイダーで次の手順で行います: [GitHub](https://vercel.com/docs/v2/git-integrations/vercel-for-github) / [GitLab](https://vercel.com/docs/v2/git-integrations/vercel-for-gitlab) / [BitBucket](https://vercel.com/docs/v2/git-integrations/vercel-for-bitbucket)).
3. 設定が完了したら、「Import Project From ...」をクリックしてNext.jsアプリをVercelにインポートします。アプリがNext.jsを使用していることが自動検出され、ビルド設定が設定されます。何も変更する必要はありません。
4. インポート後、Next.jsアプリはデプロイされ、デプロイ用URLが提供されます。「Visit」をクリックして、デプロイされたアプリを確認しましょう。

> おめでとうございます！Next.jsアプリのデプロイが完了しました。質問がある場合は、[Vercel documentation](https://vercel.com/docs) を参照してください。

Reviewer: ```suggestion
おめでとうございます！Next.jsアプリのデプロイが完了しました。疑問点がある場合は、[Vercel documentation](https://vercel.com/docs) を参照してください。
```
質問というと問い合わせを連想させるので、疑問点とするのはどうでしょう？

Reviewer 2: 確かに、「疑問点」の方が訳語として最適ですね。

> > [Next.js においてカスタマイズされたサーバー設定](/docs/advanced-features/custom-server.md)を使用したい場合は、他への移行を強くお勧めします(例えば、[ダイナミックルーティング](/docs/routing/dynamic-routes.md)を使用するなど)。移行できない場合は、[他のホスティングオプション](#other-hosting-options)を検討してください。

Reviewer: ```suggestion
> [カスタムサーバー](/docs/advanced-features/custom-server.md)を使用している場合は、代替手法への移行を強くお勧めします(例えば、[動的ルーティング](/docs/routing/dynamic-routes.md)を使用するなど)。移行できない場合は、[他のホスティング方法](#other-hosting-options)を検討してください。
```
"custom server"はシンプルにカスタムサーバーとするのはどうでしょうか？また、「他への移行」は「カスタムサーバから他の方法（動的ルーティングなど）への移行」を指していますので、わかりやすいように少し表現を変えてみました。また、ここの「オプション」も「方法」に言い換えるのはどうでしょう？

Reviewer 2: `他のホスティングオプション` → `他のホスティング方法`
おっしゃる通り、そちらの方がわかりやすいですね。

> ### DPS: 開発（Develop）、プレビュー（Preview）、出荷（Ship）

Reviewer: ```suggestion
### DPS: 開発（Develop）、プレビュー（Preview）、公開（Ship）
```
Shipを公開と訳すのはどうでしょうか？

Reviewer 2: Ship の訳語悩んだのですが、「公開」の方がしっくりする気がしますね。

> ここでは、[Vercel](https://vercel.com)が推奨するワークフローについて説明します。[Vercel](https://vercel.com)では、**DPS**ワークフローと呼ばれるものをサポートしています。**DPS**とは、開発（**D**evelop）・プレビュー（**P**review）・出荷（**S**hip）のことです。

Reviewer: ```suggestion
ここでは、[Vercel](https://vercel.com)が推奨するワークフローについて説明します。[Vercel](https://vercel.com)では、私たちが**DPS**と呼ぶワークフローを支持しています。**DPS**とは、開発（**D**evelop）・プレビュー（**P**review）・公開（**S**hip）の頭字語です。
```
"what we call"とありますので、ニュアンスを含むように修正しました！また、ここでいう"supports"は「支持する」と訳すのが良いかと思いますがどうでしょうか？

> ここでは、[Vercel](https://vercel.com)が推奨するワークフローについて説明します。[Vercel](https://vercel.com)では、いわゆる **DPS** ワークフローを支持しています。**DPS** とは、開発（**D**evelop）・プレビュー（**P**review）・公開（**S**hip）の頭字語です。

Reviewer 3: 1点細かいところですが、ここは「支持」よりも「サポート」の方が自然な訳かと思いました！

Reviewer: すみませんこちら私が直したとこでした🙏
ここのsupportは推しているとか支援しているという意味かと思うので、機能が対応しているというような意味が強い「サポートする」より「支持する」が良いかと思ったのですが、どうでしょう？

Reviewer 3: 申し訳ありません、経緯確認不足でした！確かによく読んだら、おっしゃる通り推すというニュアンスの支持の方が良さそうですね！

- **開発（Develop）:** Next.js のコードを書きましょう。開発サーバーを稼働させておき、[React Fast Refresh](https://nextjs.org/blog/next-9-4#fast-refresh)を活用しましょう。
- **プレビュー（Preview）:** GitHub / GitLab / BitBucket 上に変更をプッシュするたびに、自動的に一意の URL を持つ新しいデプロイを実施します。

> プルリクエストを開いたときに GitHub で見ることができますし、Vercel のプロジェクトページの「デプロイメントのプレビュー」で見ることもできます。[詳細はこちらをご覧ください](https://vercel.com/features/deployment-previews)。

Reviewer: ```suggestion
プルリクエストを開くと GitHub で見ることができますし、Vercel のプロジェクトページの「デプロイメントのプレビュー」で見ることもできます。[詳細はこちらをご覧ください](https://vercel.com/features/deployment-previews)。
```
少し冗長かと思い修正しましたがいかがでしょうか？

Reviewer 2: 確かに、「ときに」を多用しすぎていました:bow:

> - **出荷（Ship）:** 本番環境に出荷する準備ができたら、プルリクエストをデフォルトのブランチ (例: `master`) にマージします。Vercel は自動的に本番環境のデプロイを実施し、本番環境を更新します。

Reviewer: ```suggestion
- **公開（Ship）:** 本番環境で公開する準備ができたら、プルリクエストをデフォルトのブランチ (例: `master`) にマージします。Vercel  は自動的に本番環境へのデプロイを行います。
```
少し言い回しを変えましたのでご確認ください！

Reviewer 2: 冗長な表現でした:bow:

DPS ワークフローを使用することで、_コードレビュー_ に加えて、_デプロイ前のプレビュー_ を行うことができます。各デプロイはユニークな URL を生成し、それを共有したり、統合テストに使用したりできます。

### Next.js のための最適化

[Vercel](https://vercel.com)は、Next.jsのクリエイターが作ったもので、Next.jsを最も手厚くサポートしています。

> 例えば、[ハイブリッドページ](/docs/basic-features/pages.md)を活用する手法は完全にサポートされています。

Reviewer: ```suggestion
例えば、[ハイブリッドページ](/docs/basic-features/pages.md)については完全にサポートされています。
```
少し冗長に感じましたので修正しましたがいかがでしょうか？

> - [静的サイト生成]（/docs/basic-features/pages.md#static-generation）または[サーバサイドレンダリング]（/docs/basic-features/pages.md#server-side-rendering）を使用できます。

Reviewer: ```suggestion
- 各ページについて[静的サイト生成]（/docs/basic-features/pages.md#static-generation）または[サーバサイドレンダリング]（/docs/basic-features/pages.md#server-side-rendering）を使用できます。
```
"Every page"のニュアンスを加えました！

- [静的サイト生成]（/docs/basic-features/pages.md#static-generation）により生成される各種アセット（JS、CSS、画像、フォントなど）を使用したページは、自動的に[Vercel Smart CDN](https://vercel.com/smart-cdn)から配信されます。

> - [サーバサイドレンダリング](/docs/basic-features/pages.md#server-side-rendering) と [APIルート](/docs/api-routes/introduction.md) を使用しているページは、自動的にページと分離されたサーバレスなファンクションとしてデプロイされます。これにより、レンダリングとAPIリクエストを無限にスケールさせることができます。

Reviewer: ```suggestion
- [サーバサイドレンダリング](/docs/basic-features/pages.md#server-side-rendering) と [APIルート](/docs/api-routes/introduction.md) を使用しているページは、独立したサーバレス関数へと自動的に変換してデプロイします。これにより、レンダリングとAPIリクエストを無限にスケールさせることができます。
```
「自動的にページと分離されたサーバレスなファンクションとしてデプロイ」が冗長に感じましたので少し言い回しを変えましたがどうでしょうか？

Reviewer 2: 「ページと分離」　→　「独立」
おっしゃる通り、冗長でした:bow:

### カスタムドメイン、環境変数、自動HTTPS（SSL化） など

- **カスタムドメイン:** [Vercel](https://vercel.com) にデプロイするNext.jsアプリはカスタムドメインを割り当て可能です。[こちらのドキュメントを参照してください](https://vercel.com/docs/v2/custom-domains)。

> - **環境変数:** Vercelでは環境変数を設定できます。詳細は、[ここにあるドキュメント](https://vercel.com/docs/v2/build-step#using-environment-variables-and-secrets) を参照してください。そして、Next.jsアプリ内で[これらの環境変数](/docs/api-reference/next.config.js/environment-variables.md)を使用できます。

Reviewer: ```suggestion
- **環境変数:** Vercel では環境変数を設定できます。詳細は、[ここにあるドキュメント](https://vercel.com/docs/v2/build-step#using-environment-variables-and-secrets) を参照してください。Next.js アプリ内で[これらの環境変数](/docs/api-reference/next.config.js/environment-variables.md)を使用できます。
```
「そして」はなくても良いかと思いましたがいかがでしょうか？

- **自動HTTPS（SSL化）:** HTTPSはデフォルトで有効になっており（カスタムドメインを含む）、余分な設定は必要ありません。SSL証明書は自動で更新されます。

> - **上記以外:** Vervelというプラットフォームを知るために[ドキュメント](https://vercel.com/docs)を読んでいただければ幸いです。

Reviewer: ```suggestion
- **その他:** Vervel というプラットフォームをさらに知るために[ドキュメント](https://vercel.com/docs)をご覧ください。
```
"Read.." ですので、それほど丁寧な言い回しである必要はないと感じましたがいかがでしょうか？また、"learn more"のニュアンスを加えました！

> ## その他のホスティングオプション

Reviewer: ```suggestion
## その他のホスティング方法
```
上と同様です！

### Node.jsサーバー

Next.jsはNode.jsに対応しているホスティング環境であれば、どこにでもデプロイできます。
[カスタムサーバー](/docs/advanced-features/custom-server.md)を使用している場合はこのようなアプローチになります。

> あなたの`package.json`に`"build"`と`"start"`スクリプトが含まれていることを確認してください。

Reviewer: ```suggestion
あなたの `package.json` に `"build"` と `"start"` のスクリプトが含まれていることを確認してください。
```
助詞を足しました！

## Vercel (推奨)

> Next.js を本番環境にデプロイする最も簡単な方法は、Next.js の開発者が提供する **[Vercel platform](https://vercel.com)** を利用することです。[Vercel](https://vercel.com) は、静的サイト・JAMstack なアーキテクチャでのデプロイ、サーバーレスファンクション機能をサポートするグローバル CDN を備えたオールインワンプラットフォームです。

Reviewer: ```suggestion
Next.js を本番環境にデプロイする最も簡単な方法は、Next.js の開発者が提供する **[Vercel platform](https://vercel.com)** を利用することです。[Vercel](https://vercel.com) は、静的サイト・JAMstack なアーキテクチャでのデプロイや、サーバーレスファンクション機能をサポートするグローバル CDN を備えたオールインワンプラットフォームです。
```
細かくて恐縮ですが少し読みにくかったので修正しました🙏

Reviewer 2: 🙇

### はじめに

> はじめに、Next.js アプリを任意の Git プロバイダーにプッシュしてください。[GitHub](http://github.com/)、[GitLab](https://gitlab.com/)、[BitBucket](https://bitbucket.org/)などです。リポジトリはプライベートとパブリックのどちらでも構いません。

Reviewer: ```suggestion
はじめに、Next.js アプリを[GitHub](http://github.com/)、[GitLab](https://gitlab.com/)、[BitBucket](https://bitbucket.org/)など任意の Git プロバイダーにプッシュしてください。リポジトリはプライベートとパブリックのどちらでも構いません。
```
文章を統一できると思いますがどうでしょうか？

Reviewer 2: おっしゃる通りです:bow:
気付けず申し訳ありません。

次に、以下の手順に従ってください:

> 1. [Vercelにサインアップする](https://vercel.com/signup)（クレジットカードの登録は不要）。

Reviewer: ```suggestion
1. [Vercelにサインアップしてください](https://vercel.com/signup)（クレジットカードの登録は不要です）。
```
他と文体を合わせました！

> 2. サインアップ後に、[「Import Project」](https://vercel.com/import)ページに遷移します。「From Git Repository」でデプロイに使用する Git プロバイダーを選択し、設定をします。(設定は各プロバイダーで次の手順で行います: [GitHub](https://vercel.com/docs/v2/git-integrations/vercel-for-github) / [GitLab](https://vercel.com/docs/v2/git-integrations/vercel-for-gitlab) / [BitBucket](https://vercel.com/docs/v2/git-integrations/vercel-for-bitbucket)).

Reviewer: ```suggestion
2. サインアップすると、[Import Project](https://vercel.com/import)ページに遷移します。「From Git Repository」からデプロイに使用する[GitHub](https://vercel.com/docs/v2/git-integrations/vercel-for-github)、[GitLab](https://vercel.com/docs/v2/git-integrations/vercel-for-gitlab) 、[BitBucket](https://vercel.com/docs/v2/git-integrations/vercel-for-bitbucket))などの Git プロバイダーを選択して設定を行います。(設定は各プロバイダーで次の手順で行います。
```
こちらも気になったので直しました！

Reviewer 2: 修正ありがとうございます:bow:

`設定を行います。(設定は各プロバイダーで次の手順で行います。`
上記の箇所、そのままだと linter にも指摘されていましたので、以下のように修正しました。
`設定をします（設定は各プロバイダーで次の手順で行います）。`

### DPS: 開発（Develop）、プレビュー（Preview）、公開（Ship）

> ここでは、[Vercel](https://vercel.com)が推奨するワークフローについて説明します。[Vercel](https://vercel.com)では、私たちが**DPS**と呼ぶワークフローを支持しています。**DPS**とは、開発（**D**evelop）・プレビュー（**P**review）・公開（**S**hip）の頭字語です。

Reviewer: ```suggestion
ここでは、[Vercel](https://vercel.com)が推奨するワークフローについて説明します。[Vercel](https://vercel.com)では、いわゆる**DPS**ワークフローを支持しています。**DPS**とは、開発（**D**evelop）・プレビュー（**P**review）・公開（**S**hip）の頭字語です。
```
申し訳ありません！"what we call"は「いわゆる」と訳す方が自然だと思い至りました🙏（DPSはVercel以外では聞かないような気がするので、修正前の訳でも良い気はしますが・・）

- [静的サイト生成]（/docs/basic-features/pages.md#static-generation）により生成される各種アセット（JS、CSS、画像、フォントなど）を使用したページは、自動的に[Vercel Smart CDN](https://vercel.com/smart-cdn)から配信されます。
- [サーバサイドレンダリング](/docs/basic-features/pages.md#server-side-rendering) と [APIルート](/docs/api-routes/introduction.md) を使用しているページは、独立したサーバレス関数へと自動的に変換してデプロイします。これにより、レンダリングと API リクエストを無限にスケールさせることができます。

### カスタムドメイン、環境変数、自動HTTPS（SSL化）など

- **カスタムドメイン:** [Vercel](https://vercel.com) にデプロイする Next.js アプリはカスタムドメインを割り当て可能です。[こちらのドキュメントを参照してください](https://vercel.com/docs/v2/custom-domains)。
- **環境変数:** Vercel では環境変数を設定できます。詳細は、[ここにあるドキュメント](https://vercel.com/docs/v2/build-step#using-environment-variables-and-secrets) を参照してください。Next.js アプリ内で[これらの環境変数](/docs/api-reference/next.config.js/environment-variables.md)を使用できます。
- **自動 HTTPS（SSL 化）:** HTTPS はデフォルトで有効になっており（カスタムドメインを含む）、余分な設定は必要ありません。SSL 証明書は自動で更新されます。
- **その他:** Vervel というプラットフォームをさらに知るために[ドキュメント](https://vercel.com/docs)をご覧ください。

## その他のホスティング方法

### Node.jsサーバー

Next.js は Node.js に対応しているホスティング環境であれば、どこにでもデプロイできます。
[カスタムサーバー](/docs/advanced-features/custom-server.md)を使用している場合はこのようなアプローチになります。

> あなたの `package.json` に `"build"` と `"start"` のスクリプトが含まれていることを確認してください。

Reviewer: ```suggestion
`package.json` に `"build"` と `"start"` のスクリプトが含まれていることを確認してください。
```
「あなたの」はあえてつけなくて良いかと思いました

Reviewer 2: 🙇

}
```

`next build`スクリプトは `.next` フォルダ内に本番用アプリケーションをビルドします。

> ビルド後、`next start`スクリプトにより [Pagesにおける動的なルーティング](/docs/basic-features/pages.md) をサポートする Node.js サーバを起動します。

Reviewer: ```suggestion
ビルド後に実行する `next start` により、静的に生成されたページとサーバサイドでレンダリングされたページ の[ハイブリッドページ](/docs/basic-features/pages.md) をサポートする Node.js サーバを起動します。
```
下の文章と統合してみましたがいかがでしょうか？

> 静的に生成されたページとサーバサイドでレンダリングされたページの両方をサポートします。

Reviewer: ```suggestion
```
統合したのでここは削除します！

> ### 静的サイト生成

Reviewer: ```suggestion
### 静的 HTML の出力
```
Exportですので作成ではなく出力（あるいは単にエクスポート）が良いかと思いますがいかがでしょう？

> Next.jsアプリの静的サイト生成を用いたい場合は、[ドキュメント](/docs/advanced-features/static-html-export.md)の指示に従ってください。デフォルトでは、`next export`によりは`out`ディレクトリに、CDN や静的サイトホスティングサービスで用いることができる静的サイトを生成します。

Reviewer: ```suggestion
Next.js アプリを静的 HTML へと出力したい場合は、[ドキュメント](/docs/advanced-features/static-html-export.md)の指示に従ってください。デフォルトでは、`next export`によって`out`ディレクトリが生成され、 CDN や静的サイトホスティングサービスで配信することができます。
```
言い回し修正しましたのでご確認ください！

> > ご利用のNext.jsアプリが完全に静的サイトの場合でも、[Vercel](https://vercel.com/)を使用することを強くお勧めします。[Vercel](https://vercel.com/) は、静的な Next.js アプリが非常に高速に動作するように最適化されています。`next export`で生成された静的サイトは、Vercel では設定を一切行わずにデプロイすることでも動作します。

Reviewer: ```suggestion
> ご利用の Next.js アプリが完全に静的サイトの場合でも、[Vercel](https://vercel.com/) を使用することを強くお勧めします。[Vercel](https://vercel.com/) は、静的な Next.js アプリが非常に高速に動作するように最適化されています。`next export` は、Vercel では設定を一切行わずにデプロイしても動作します。
```
「することでも」が少し冗長に感じましたが、いかがでしょうか？
